From 4f87c61f07ca4c692d9a5b1f1c538396857ad112 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Thu, 13 Jan 2011 13:23:30 +0800
Subject: [PATCH] ENGR00137927-3 usb-host add vbus ON operation for OTG host driver

The old implementation did not cover the host only driver for
OTG port situation.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 drivers/usb/host/ehci-arc.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/host/ehci-arc.c b/drivers/usb/host/ehci-arc.c
index 7764b10..af86a96 100644
--- a/drivers/usb/host/ehci-arc.c
+++ b/drivers/usb/host/ehci-arc.c
@@ -242,7 +242,8 @@ int usb_hcd_fsl_probe(const struct hc_driver *driver,
 		retval = otg_set_host(ehci->transceiver, &ehci_to_hcd(ehci)->self);
 		if (retval)
 			otg_put_transceiver(ehci->transceiver);
-	} else if (pdata->operating_mode == FSL_USB2_MPH_HOST)
+	} else if ((pdata->operating_mode == FSL_USB2_MPH_HOST) || \
+			(pdata->operating_mode == FSL_USB2_DR_HOST))
 		fsl_platform_set_vbus_power(pdata, 1);
 
 	if (pdata->suspended) {
-- 
1.5.4.4

